manager/manager.go

```diff
@@ -75,9 +92,27 @@ func (rm *realManager) Housekeep() {
 }
 
 func (rm *realManager) housekeep(start, end time.Time) {
-	data := rm.sources.ScrapeSources(start, end)
-	for _, p := range rm.processors {
-		data = p.Process(data)
+	if !start.Before(end) {
```

Comment:
This is already enforced in Housekeep. Simple sanity check is probably more than enough.